scripts/start.py

```diff
@@ -156,65 +146,64 @@ def main(args=None):
+        automatic_restart=not parsed_args.no_auto_restart,
+        skip_sdk_update_check=True, port=PORT_NUMBER_FOR_GAE_SERVER)
 
-    # Launch a browser window.
-    if common.is_linux_os() and not parsed_args.no_browser:
-        detect_virtualbox_pattern = re.compile('.*VBOX.*')
-        if list(filter(
-                detect_virtualbox_pattern.match,
-                os.listdir('/dev/disk/by-id/'))):
-            common.print_each_string_after_two_new_lines([
-                'INFORMATION',
-                'Setting up a local development server. You can access this '
-                'server',
-                'by navigating to localhost:%s in a browser window.'
-                % python_utils.UNICODE(PORT_NUMBER_FOR_GAE_SERVER)])
-        else:
+    python_utils.PRINT('Starting GAE development server')
+    with managed_dev_appserver:
+        # Wait for the servers to come up.
+        while not common.is_port_open(PORT_NUMBER_FOR_GAE_SERVER):
+            time.sleep(1)
+
+        # Launch a browser window.
+        if common.is_linux_os() and not parsed_args.no_browser:
+            detect_virtualbox_pattern = re.compile('.*VBOX.*')
+            if list(filter(
+                    detect_virtualbox_pattern.match,
+                    os.listdir('/dev/disk/by-id/'))):
+                common.print_each_string_after_two_new_lines([
+                    'INFORMATION',
+                    'Setting up a local development server. You can access '
+                    'this server',
+                    'by navigating to localhost:%s in a browser window.'
+                    % python_utils.UNICODE(PORT_NUMBER_FOR_GAE_SERVER)])
+            else:
+                common.print_each_string_after_two_new_lines([
+                    'INFORMATION',
+                    'Setting up a local development server at localhost:%s. '
+                    % python_utils.UNICODE(PORT_NUMBER_FOR_GAE_SERVER),
+                    'Opening a default browser window pointing to this server'])
+                time.sleep(5)
+                background_processes.append(
+                    subprocess.Popen([
+                        'xdg-open', 'http://localhost:%s/'
+                        % python_utils.UNICODE(PORT_NUMBER_FOR_GAE_SERVER)]))
+        elif common.is_mac_os() and not parsed_args.no_browser:
             common.print_each_string_after_two_new_lines([
                 'INFORMATION',
                 'Setting up a local development server at localhost:%s. '
                 % python_utils.UNICODE(PORT_NUMBER_FOR_GAE_SERVER),
-                'Opening a default browser window pointing to this server'])
+                'Opening a default browser window pointing to this server.'])
             time.sleep(5)
             background_processes.append(
                 subprocess.Popen([
-                    'xdg-open', 'http://localhost:%s/'
+                    'open', 'http://localhost:%s/'
```

Comment:
Did you test this? (Looked this up and seems that `open` is indeed MacOS command, just want to make sure that this was tested)

Reply:
Ah looks like this is a diff-algorithm issue, this is actually the same (see line 183 above this)

Reply:
Actually, I think I really did let a bug go through (unrelated, but fixed). Marking as resolved!